modules/upgrade/pages/proxy-uyuni.adoc

```diff
@@ -0,0 +1,51 @@
+[[proxy-uyuni-upgrade]]
+= Proxy - Upgrade Procedure
+
+Before you perform any proxy update, schedule a maintenance window.
+The clients registered to {productname} through the proxy will not be able to connect to {productname} while the update is in progress.
+For more information about maintenance windows, see xref:administration:maintenance-window.adoc[].
+
+
+
+== Upgrade the Proxy
+
+To upgrade a proxy you first stop the proxy service, then you replace the software repositories and update the software, and finally you restart the proxy service.
+
+
+
+.Procedure: Updating the {productname} Proxy
+. On the {productname} Proxy, stop the proxy service:
++
+----
+spacewalk-proxy stop
+----
+
+// there is no chance to fix any issues during migration,
+// make sure you have a backup before continuing. If you are
+// running Uyuni server on a virtual machine, it is advisable
+// to create a snapshot before performing the migration!
+
+. Replace and refresh software repositories:
++
+----
+mv /etc/zypp/repos.d /etc/zypp/repos.d.old
```

Comment:
Huh? what's this :-?

Reply:
It's creating a backup of the old repos directory but this is wrong. What if the user has some other repositories enabled?

Reply:
On a proxy, it should not. This is not what worries me most. The whole script is what worries me. I think it's the wrong way to do the uprade.

Reply:
Funny enough it seems we tell people to install the Uyuni Proxy not as a client: https://www.uyuni-project.org/uyuni-docs/uyuni/installation/install-proxy-uyuni.html

I don't know why this is this way. To me it seems strange, but ok. For now it is how it is. So the script seems to make sense.

Reply:
Ah *now* I understand it. You tell them to it that way, but then you say this: https://www.uyuni-project.org/uyuni-docs/uyuni/installation/uyuni-proxy-registration.html (BTW the channel name for `spacewalk-common-channels` is not `uyuni-proxy-stable-leap-151` bot `uyuni-proxy-stable-leap`)

So in the end this migration script is still wrong, as the proxies will be clients. If you run this script, yeah, the migration will work, but then the Server will still think the proxy is a Leap 15.1, so I guess it will crate a mess.

This what I suspect that needs to be done if you want to migrate:

1. Add the Leap 15.2 channels with `spacewalk-common-channels opensuse_leap15_2  opensuse_leap15_2-non-oss  opensuse_leap15_2-non-oss-updates  opensuse_leap15_2-updates  opensuse_leap15_2-uyuni-client uyuni-proxy-stable-leap-152`
2. Sync them: `spacewalk-repo-sync -p opensuse_leap15_2-x86_64` (this will sync the parent and all children).
3. Adjust the proxies using SSM or one by one to use the Leap 15.2 channels you just added and synced. Remember to apply the changes.
4. Perform the migration (this is the "gray" area for me, maybe this where they need to run `spacewalk-proxy stop; zypper ref; zypper -n dup --allow-vendor-change; shutdown -r now` directly on the proxies).

And that should be all.

BTW I just noticed we don't have `uyuni-server-stable-leap-152`, so time to create it.

Reply:
https://github.com/uyuni-project/uyuni/pull/2432

Reply:
OK, I thought a little about all this mess. Let's do the following:

1. I will tag Uyuni tomorrow, and will no accept more submissions to the master branch (meanwhile I will prepare a MU as well)
2. @keichwa and @mantel can test the migration of the proxy tomorrow and Thursday and have the doc ready by end of Thursday
3. @jcayouette prepares the package and the PR for the website on Thursday by the end of the day, or Friday, early in the morning.
4. I start the release on Friday morning, as soon as the PR and the package for the doc are ready.

Sounds fair enough? That gives some more time to test the migration without that much pressure.

Please ACK tomorrow morning if we have an agreement.

Reply:
I answered too quickly yesterday. While this script probably would (sort of) work, the clean way to upgrade the proxy would be to initiate an autoyast upgrade from the server. This way all the information regarding the proxy would stay consistent. We even had this documented some time ago; biggest issue is providing a working autoyast file which is a bit tricky.

Reply:
> So in the end this migration script is still wrong, as the proxies will be clients. If you run this script, yeah, the migration will work, but then the Server will still think the proxy is a Leap 15.1, so I guess it will crate a mess.

Maybe this could even be fixed up by refreshing the information from the proxy, but yes, the only really clean way is to perform the upgrade via autoyast so the server is aware of any modifications done to the proxy.

Reply:
I think that providing manual instructions should be fine as well (if it is easier) as long as they are accurate. Something like what I described above.

Reply:
> I think that providing manual instructions should be fine as well (if it is easier) as long as they are accurate. Something like what I described above.

Yes, especially as it is much easier than to fiddle around with autoyast files. We just need to confirm that the changes in the proxy's system can be updated to the new state afterwards.